test/input-states-mocha.js

```diff
@@ -0,0 +1,333 @@
+          inState = { vnid: 22,
+                      data: 'middle data',
+                      lm: createLm() };
+          node.vni().inputStates({'input1': inState});
+          inStates.push( { portInfo: inPortInfo, state: inState } ); 
+
+          // Verify that we have all of them 
+          inStates.forEach( function(inState) { 
+              var actualState = node.vni().inputStates(inState.portInfo.inportName);
+              actualState.should.deep.equal( inState.state );
+          });
+    });
+    it('should fall back to no vnid state', function() {
+         var state = { vnid: '010',
+                       data: 'hello world from originating node',
+                       lm: createLm() }
+
+         var context = node.vni().inputStates('inputa', 0, state);
+         context.should.exist;
+
+         var result = node.vni('010').inputStates('inputa', 0); 
+         verifyInputState( result, state );
+    }); 
+    it('should set and get a new packet input state', function() {
+         var state = { vnid: '010',
+                       data: 'hello world from originating node',
+                       lm: createLm() }
+
+         var context = node.vni().inputStates('inputa', 0, state);
+         context.should.exist;
+
+         var result = node.vni().inputStates('inputa', 0); 
+         verifyInputState( result, state );
+    }); 
+    it('should set and get a new packet input state by socketIndex', function() {
+         var state = { vnid: '010',
+                       data: 'hello world from originating node',
+                       lm: createLm() }
+
+         var context = node.vni().inputStates('inputa', 0, state);
+         context.should.exist;
+
+         verifyInputState(node.vni().inputStates('inputa', 0), state);
+         node.vni().inputStates('inputa', 0, undefined);
+         expect(node.vni().inputStates('inputa', 0)).to.be.undefined;
+    }); 
+    it('should set and get array of input states', function() {
+         var one = { vnid: '010',
+                       data: 'hello world from node one',
+                       lm: createLm() }
+         var two = { vnid: '010',
+                       data: 'hello world from node two',
+                       lm: createLm() }
+
+         var context = node.vni().inputStates({'inputa': [one, two]});
+         context.should.exist;
+
+         node.vni().inputStates('inputa').should.eql([one, two]);
+         verifyInputState( node.vni().inputStates('inputa', 0), one );
+         verifyInputState( node.vni().inputStates('inputa', 1), two );
+    }); 
+    it('should set and get hash of array of input states', function() {
+         var one = { vnid: '010',
+                       data: 'hello world from node one',
+                       lm: createLm() }
+         var two = { vnid: '010',
+                       data: 'hello world from node two',
+                       lm: createLm() }
+
+         var context = node.vni().inputStates({inputa: [one, two]});
```

Comment:
I get that this data is supposed to be same vnid from two different nodes, coming into inputa, but I don't see anything set that would ensure input-state treats it that way.    This looks more like a sequence of IIPs with the same vnid where is the set up if this is for multiple source nodes, where is that set up/defined.  Some commenting here might help.